jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java

```diff
@@ -145,38 +145,33 @@ private void updateImageName(KubernetesClient kubernetes, String namespace, HasM
         if (entity instanceof Deployment) {
             Deployment resource = (Deployment) entity;
             DeploymentSpec spec = resource.getSpec();
-            if (spec != null) {
-                if (updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
-                    kubernetes.apps().deployments().inNamespace(namespace).withName(name).replace(resource);
-                }
+            if (spec != null && updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
+                kubernetes.apps().deployments().inNamespace(namespace).withName(name).replace(resource);
+                kubernetes.apps().deployments().inNamespace(namespace).withName(name).rolling().restart();
```

Comment:
Umm, why is this required? As far as I remember `replace()` also triggers rolling update

Reply:
It doesn't, at least in this version. Check the code, maybe I missed something.

Reply:
ohk, I tested it and looks like image doesn't get updated with a single `replace()`.

Reply:
We should revisit this once we update Kubernetes Client. If it's still not working, maybe evaluate alternatives and improve Kubernetes Client for this scenario.